File: core/src/main/kotlin/br/com/dillmann/nginxignition/core/accesslist/AccessListRepository.kt
package br.com.dillmann.nginxignition.core.accesslist

import br.com.dillmann.nginxignition.core.common.pagination.Page
import java.util.UUID

interface AccessListRepository {
    suspend fun findById(id: UUID): AccessList?
    suspend fun findByName(name: String): AccessList?
    suspend fun findPage(pageNumber: Int, pageSize: Int, searchTerms: String?): Page<AccessList>
    suspend fun findAll(): List<AccessList>
    suspend fun save(accessList: AccessList)
    suspend fun deleteById(id: UUID)
}

File: core/access_list/access_list_repository.go
package access_list

import (
	"dillmann.com.br/nginx-ignition/core/common/pagination"
	"github.com/google/uuid"
)

type AccessListRepository interface {
	FindById(id uuid.UUID) (*AccessList, error)
	DeleteById(id uuid.UUID) error
	FindByName(name string) (*AccessList, error)
	FindPage(pageNumber int64, pageSize int64, searchTerms string) (*pagination.Page[AccessList], error)
	FindAll() (*[]AccessList, error)
	Save(accessList *AccessList) error
}
